File: insuranceclaim.client/src/app/Models/ReserveModel.ts
export interface IReserveRequestModel {
  id: number,
  reserveDamage: number,
  reserveClaimantCost: number,
  reserveDefenceCost: number,

}


export interface IReserveModel {
  id: number,
  reserveDamage: number,
  reserveClaimantCost: number,
  reserveDefenceCost: number,
  paidDamage: number,
  paidClaimantCost: number,
  paidDefenceCost: number,
  incurredDamage: number,
  incurredClaimantCost: number,
  incurredDefenceCost: number,
  isInApproval: boolean,
  isOverRidden:boolean,
  status: string,
  statusDate: Date,
}

export interface IReserveModel1 extends IReserveModel{
  statusDate1: Date,

}
export interface DataObject {
  date: Date;
  [key: string]: any;  // Allow other properties in the object
}

File: InsuranceClaim.Server/Models/ReserveModel.cs
namespace InsuranceClaim.Server.Models
{
    public class ReserveModel
    {
        public int Id { get; set; }
        public double ReserveDamage { get; set; }
        public double ReserveClaimantCost { get; set; }
        public double ReserveDefenceCost { get; set; }
        public double PaidDamage { get; set; }
        public double PaidClaimantCost { get; set; }
        public double PaidDefenceCost { get; set; }
        public double IncurredDamage { get; set; }
        public double IncurredClaimantCost { get; set; }
        public double IncurredDefenceCost { get; set; }

        public Boolean IsInApproval { get; set; }
        public string Status { get; set; }
        public DateTime StatusDate { get; set; }
        public Boolean IsOverRidden { get; set; }

        //public ReserveModel() { }

    }

    public class AddReserveRequest
    {
        public int Id { get; set; }
        public double ReserveDamage { get; set; }
        public double ReserveClaimantCost { get; set; }
        public double ReserveDefenceCost { get; set; }

    }

}
